src/Cryptography/Sodium/Bindings/SHA512.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE Trustworthy #-}
+
+-- |
+--
+-- Module: Cryptography.Sodium.Bindings.SHA512
+-- Description: Direct bindings to the public-key signing algorithm ed25519 implemented in Libsodium
+-- Copyright: (C) Hécate Moonlight 2022
+-- License: BSD-3-Clause
+-- Maintainer: The Haskell Cryptography Group
+-- Stability: Stable
+-- Portability: GHC only
+module Cryptography.Sodium.Bindings.SHA512
+  ( -- * Introduction
+    -- $introduction
+    CryptoHashSHA512State (..),
+    cryptoHashSHA512Bytes,
+    cryptoHashSHA512,
+    cryptoHashSHA512Init,
+    cryptoHashSHA512Update,
+    cryptoHashSHA512Final,
+  )
+where
+
+import Foreign (Ptr, Word64, Word8)
+import Foreign.C (CInt (..), CSize (..), CUChar (..), CULLong (..))
+
+-- $introduction
+--
+-- The SHA-512 functions are provided for interoperability with other applications. If you are
+-- looking for a generic hash function and not specifically SHA-2, using
+-- 'Cryptography.Sodium.Bindings.GenericHashing' (BLAKE2b) might be a better choice. These functions
+-- are also not suitable for hashing passwords or deriving keys from passwords. Use
+-- 'Cryptography.Sodium.Bindings.PasswordHashing' instead.
+--
+-- These functions are not keyed and are thus deterministic. In addition, the untruncated versions
+-- are vulnerable to length extension attacks. A message can be hashed in a single pass, but a
+-- streaming API is also available to process a message as a sequence of multiple chunks.
+
+data CryptoHashSHA512State = CryptoHashSHA512State
```

Comment:
Does this correspond to anything? If so, it should have a Haddock with a link, along with an explanation.

This applies to multiple other entries in this module.